HOW_CHIALISP_IS_COMPILED.md

```diff
@@ -0,0 +1,798 @@
+Compiler theory of operation
+==
+
+Basic structure and where to look for specific things
+--
+
+_About compilers in general_
+
+What's been suggested is an overview of code compilation overall, which includes
```

Comment:
I'd think the target of this document would be someone who wants to hack on or understand PRs for the `clvm_tools_rs` repo (that's my motivation at least). I'd expect someone in that situation to know about compilers and text files, etc.